compiler/src/iree/compiler/Codegen/Common/TransformExtensions/CommonExtensions.cpp

```diff
@@ -1367,5 +1367,176 @@ void transform_dialect::IREEEraseHALDescriptorTypeFromMemRefOp::getEffects(
+  OpBuilder::InsertionGuard g(rewriter);
+  rewriter.setInsertionPoint(workgroupCountOp);
+  loc = workgroupCountOp.getLoc();
+  for (auto indices : dimCollapseMapping) {
+    // If there is a single index we can just push the corresponding workload
+    // arg.
+    if (indices.size() == 1) {
+      auto index = indices[0];
+      pushRange(newWorkload, workloadDims[index], workloadDims[index + 1]);
+      continue;
+    }
+
+    assert(indices.size());
+    SmallVector<AffineExpr> syms(indices.size());
+    bindSymbolsList(rewriter.getContext(), MutableArrayRef<AffineExpr>{syms});
+    SmallVector<Value> originalSizes;
+    AffineExpr product = syms[0];
+    SmallVector<Value> skipped;
+    for (auto [enIndex, dimIndex] : llvm::enumerate(indices)) {
+      originalSizes.push_back(workload[workloadDims[dimIndex]]);
+      if (enIndex > 0) {
+        // Push any skipped workload values.
+        pushRange(skipped, workloadDims[dimIndex - 1] + 1,
+                  workloadDims[dimIndex]);
+        product = product * syms[enIndex];
+      }
+    }
+
+    // Make+push the collapsed workload value, followed by all values
+    // until the next dim affected by the collapse_shape op.
+    auto m = AffineMap::get(0, indices.size(), product);
+    auto collapsedIndex =
+        makeComposedAffineApply(rewriter, loc, m, originalSizes);
+    newWorkload.push_back(collapsedIndex);
+    newWorkload.append(skipped);
+    pushRange(newWorkload, workloadDims[indices.back()] + 1,
+              workloadDims[indices.back() + 1]);
+  }
+
+  rewriter.replaceOpWithNewOp<IREE::Flow::DispatchWorkgroupCountFromDagRootOp>(
+      workgroupCountOp, newWorkload);
+  return success();
+}
+
+DiagnosedSilenceableFailure
+transform_dialect::ConvertConv2DToImg2ColAndAdjustWorkgroupCountOp::applyToOne(
+    linalg::LinalgOp target, transform::ApplyToEachResultList &results,
+    transform::TransformState &state) {
+  auto funcOp = target->getParentOfType<func::FuncOp>();
+
+  // Get the output map for the target operation before rewriting.
+  AffineMap outputMap = target.getIndexingMapsArray().back();
+
+  IRRewriter rewriter(target->getContext());
+  rewriter.setInsertionPoint(target);
+  // TODO: Extend this to other convolution cases by handling cases beyond
+  // collapse -> matmul -> expand from the patterns for 2D convolutions handled
+  // here.
+  auto maybeTransformed =
+      TypeSwitch<Operation *, FailureOr<std::pair<Operation *, Operation *>>>(
+          target)
+          .Case([&](linalg::Conv2DNhwcHwcfOp op) {
+            return rewriteInIm2Col(rewriter, op);
+          })
+          .Case([&](linalg::Conv2DNchwFchwOp op) {
+            return rewriteInIm2Col(rewriter, op);
+          })
+          .Default([&](Operation *op) {
+            return rewriter.notifyMatchFailure(op, "not supported");
+          });
+  if (failed(maybeTransformed)) return emitDefaultSilenceableFailure(target);
+
+  auto im2col = maybeTransformed->first;
```

Comment:
nit: spell out type here and in other places following this:
https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable